share/spack/gitlab/cloud_e4s_pipelines/.gitlab-ci.yml

```diff
@@ -1,60 +1,116 @@
-stages: [generate, build]
+stages: [ "generate", "build" ]
+
+default:
+  image: { "name": "ghcr.io/scottwittenburg/ecpe4s-ubuntu18.04-runner-x86_64:2020-09-01", "entrypoint": [""] }
+
+########################################
+# Job templates
+########################################
+
+.pr:
+  only:
+  - /^github\/pr[\d]+_.*$/
+  variables:
+    SPACK_PR_BRANCH: ${CI_COMMIT_REF_NAME}
+    SPACK_IS_PR_PIPELINE: "True"
+
+.develop:
+  only:
+  - /^github\/develop$/
+  variables:
+    SPACK_IS_PR_PIPELINE: "False"
 
 .generate:
   stage: generate
   script:
     - . "./share/spack/setup-env.sh"
     - spack --version
-    - cd share/spack/gitlab/cloud_e4s_pipelines/stacks/e4s
+    - cd share/spack/gitlab/cloud_e4s_pipelines/stacks/${SPACK_CI_STACK_NAME}
```

Comment:
What do you think of changing this path component to `gitlab_cloud_pipelines` (or something) and renaming the directory these files live in?

Reply:
I like it.  I didn't want to change any file locations though in case it broke something e4s was doing.  It definitely makes sense to generalize the directory name too though.  I'll update accordingly.